go/substrait/gen-proto.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
```

Comment:
I don't think this script is necessary if you use `buf gen`.

Reply:
Updated `buf.gen.yaml` and removed this script.